lib/config/configuration.go

```diff
@@ -1208,17 +1209,43 @@ func applyWindowsDesktopConfig(fc *FileConfig, cfg *service.Config) error {
 	}
 	ldapPassword, err := os.ReadFile(fc.WindowsDesktop.LDAP.PasswordFile)
 	if err != nil {
-		return trace.WrapWithMessage(err, "loading LDAP password from file %v",
+		return trace.WrapWithMessage(err, "error loading LDAP password from file %v",
 			fc.WindowsDesktop.LDAP.PasswordFile)
 	}
+
+	// If a CA file is provided but InsecureSkipVerifyCA is also set to true, throw an
+	// error to make sure the user isn't making a critical security mistake (i.e. thinking
+	// that their LDAPS connection is being verified to be with the CA provided, but it isn't
+	// due to InsecureSkipVerifyCA == true ).
+	if fc.WindowsDesktop.LDAP.DerEncodedCACert != "" && fc.WindowsDesktop.LDAP.InsecureSkipVerifyCA {
+		return trace.Wrap(trace.BadParameter(
+			`a CA file was provided but insecure_skip_verify_ca was also set to true;
```

Comment:
1. Do we really want a newline in the error message? Any why `trace.Wrap`? Let's just:
```
return trace.BadParameter("a CA file was provided for LDAP, but insecure_skip_verify was enabled;" +
    "remove the der_ca_file from the configuration to skip TLS verification"
```

2. I think `insecure_skip_verify` is a better name than `insecure_skip_verify_ca`. It matches the Go field of the same name and is more accurate.

3. Is the configuration field a file or the actual CA certificate contents? The field is named `DerEncodedCACert` but in the error message you refer to a file (`der_ca_file`).

4. DER is an acronym, so `DEREncodedCACert` is a better name.